include/swift/SIL/SILInstruction.h

```diff
@@ -6938,6 +7181,34 @@ SILFunction *ApplyInstBase<Impl, Base, false>::getCalleeFunction() const {
   }
 }
 
+class DestructureValueInst final
+    : public UnaryInstructionBase<SILInstructionKind::DestructureValueInst,
+                                  MultipleValueInstruction>,
+      private llvm::TrailingObjects<DestructureValueInst,
+                                    MultipleValueInstructionResult> {
+  friend TrailingObjects;
+
+  DestructureValueInst(SILDebugLocation Loc, SILValue Operand)
+      : UnaryInstructionBase(Loc, Operand) {}
+
+public:
+  static DestructureValueInst *create(SILDebugLocation Loc, SILFunction &F,
+                                      SILValue Operand);
+};
+
+inline SILNode *MultipleValueInstructionResult::getCanonicalSILNodeInObject() {
+  return Parent->getCanonicalSILNodeInObject();
+}
+
+inline const SILNode *
+MultipleValueInstructionResult::getCanonicalSILNodeInObject() const {
+  return Parent->getCanonicalSILNodeInObject();
+}
+
+inline unsigned MultipleValueInstructionResult::getIndex() const {
+  return Parent->getIndexOfResult(this);
+}
```

Comment:
Why are these defined down here instead of after MultiValueInstruction?

Reply:
IIRC the problem was forward declarations not being defined yet.